### PR TITLE
fix for #751

### PR DESCRIPTION
**Pull request by ramusus**

### CODE REVIEW

File: packages/reaction-core/common/router.js [hunk 1]
```diff
@@ -156,7 +156,10 @@ Router.map(function () {
     name: "index",
     template: "products",
     subscriptions: function () {
-      this.subscribe("Products", Session.get("productScrollLimit"));
+      Session.set("productFilters", {});
+      Deps.autorun(function() {
```
Comment (zenweasel) [inline packages/reaction-core/common/router.js]: Missing space

File: packages/reaction-core/common/router.js [hunk 2]
```diff
@@ -230,22 +233,14 @@ Router.map(function () {
...
-        return {
-          tag: ReactionCore.Collections.Tags.findOne({
-            slug: id
-          }) || ReactionCore.Collections.Tags.findOne(id)
-        };
-      }
+      const id = this.params._id;
+      const tag = ReactionCore.Collections.Tags.findOne(id) || ReactionCore.Collections.Tags.findOne({slug: id});
+      Session.set("productFilters", {"tag": tag._id});
+      Deps.autorun(function() {
```
Comment (zenweasel) [inline packages/reaction-core/common/router.js]: Missing space

**Comment by zenweasel** [comment]
@ramsus Are you running ESLint with our `.eslintrc`? This would catch all the nit-picky format things.

File: packages/reaction-collections/server/publications/products.js [hunk 3]
```diff
@@ -15,24 +15,30 @@ Meteor.publish("Products", function (productScrollLimit, shops) {
...
-      for (let thisShopId of shops) {
-        if (Roles.userIsInRole(this.userId, ["admin", "createProduct"],
-            thisShopId)) {
-          shopAdmin = true;
+    let selector = {shopId: shop._id};
+
+    if (productFilters) {
+      // handle multiple shops
+      if (productFilters.shops) {
+        check(productFilters.shops, Array);
```
Comment (newsiberian) [inline packages/reaction-collections/server/publications/products.js]: hello,
maybe:  `if (Array.isArray(productFilters.shops)) {`?
and line 38: `if (typeof productFilters.tag === "string") {`?

did you test with both `shops` and `tag` provided. Is request valid without [AND](https://docs.mongodb.org/manual/reference/operator/query/and/)?

Comment (ramusus) [inline packages/reaction-collections/server/publications/products.js]: I think we should check format of this argument as usual we do in the top of all publications.
Yes, I checked, AND is not necessary here

**Comment by ramusus** [comment]
@zenweasel, installed eslint to webstorm, thanks for suggestion!

**Comment by zenweasel** [comment]
@ramusus I found you needed to toggle "Prefer Strict Mode" in the Javascript pane to get ESLint to activate in WebStorm after installing. At least for me.

**Comment by ramusus** [comment]
@zenweasel yes I did this, do you reformat code every time manually before saving this file?

**Comment by zenweasel** [comment]
@ramsus Rarely need to reformat, and never need to save when using WebStorm. It saves automatically.

**Comment by zenweasel** [comment]
@ramusus I am seeing an issue when testing this PR. If I follow the instructions for duplicating the error it appears that the product grid is no longer filtering by tag. For the second tag I created I get all products, even those without this tag.

**Comment by aaronjudd** [comment]
Approved! this is most likely to get refactored in v0.12, but we'll see how this merges

**Pull request merged**